ion-schema/src/isl/isl_constraint.rs

```diff
@@ -433,12 +439,37 @@ pub mod v_2_0 {
         )
     }
 
-    /// Creates an `annotations` constraint using [str]s and [Element]s specified inside it
-    pub fn annotations<'a, A: IntoIterator<Item = &'a str>, B: IntoIterator<Item = Element>>(
-        annotations_modifiers: A,
-        annotations: B,
+    /// Creates an `annotations` constraint using a list of valid annotations and specify whether annotations are required or closed or both.
+    pub fn simple_annotations<A: IntoIterator<Item = Element>>(
+        is_required: bool,
+        is_closed: bool,
+        annotations: A,
     ) -> IslConstraint {
-        todo!()
+        let annotations: Vec<Annotation> = annotations
+            .into_iter()
+            .map(|a| {
+                Annotation::new(
+                    a.as_text().unwrap().to_owned(),
+                    Annotation::is_annotation_required(&a, is_required),
+                )
+            })
+            .collect();
+        IslConstraint::new(
+            IslVersion::V2_0,
+            IslConstraintImpl::Annotations(IslAnnotationsConstraint::SimpleAnnotations(
+                IslSimpleAnnotationsConstraint::new(is_closed, false, is_required, annotations),
+            )),
+        )
+    }
+
+    /// Creates an `annotations` constraint using an [IslTypeRef].
+    pub fn standard_annotations(isl_type: IslTypeRef) -> IslConstraint {
```

Comment:
minor naming suggestion—if it's "standard", then we probably don't need to qualify the name.
```suggestion
    pub fn annotations(isl_type: IslTypeRef) -> IslConstraint {
```